types/notifications.go

```diff
@@ -13,3 +18,60 @@ const (
 	FlappingStart
 	FlappingEnd
 )
+
+/*
+ * Parse a notification type string (typically received when Icinga 2 is executing a notification plugin)
+ * into a fitting enum to simplify the following logic.
+ */
+func ParseNotificationType(nt string) (NotificationType, error) {
+	switch strings.ToLower(nt) {
+	case "downtimestart":
+		return DowntimeStart, nil
+	case "downtimeend":
+		return DowntimeEnd, nil
+	case "downtimeremoved":
+		return DowntimeRemoved, nil
+	case "custom":
+		return Custom, nil
+	case "acknowledgement":
+		return Acknowledgement, nil
+	case "problem":
+		return Problem, nil
+	case "recovery":
+		return Recovery, nil
+	case "flappingstart":
+		return FlappingStart, nil
+	case "flappingend":
+		return FlappingEnd, nil
+	default:
+		return 0, errors.New("no matching state for the provided string")
+	}
+}
+
+/*
+ * Reverse of the above function for completeness
```

Comment:
```suggestion
 * Transforms a notification type into a string
```

Also, this might be a job for the Stringer interface.

```
func (n NotificationType) String() string {
	switch n {
	case DowntimeStart:
		return "downtimestart"
...
```